finetuner-workflow/finetuner/finetuner.py

```diff
@@ -4,37 +4,53 @@
 import json
 import gc
 import resource
+import mmap
+
+import numpy
 import psutil
 import pynvml
 import os
 import sys
 import struct
-import socket
+import time
 import math
 import torch
+from torch import Tensor
 from torch.utils.data import Dataset, random_split, RandomSampler
 import argparse
 import pathlib
-from typing import Callable, Tuple
-try:
-    from tensorizer.tensorizer import load_model, get_tokenizer, no_init
-except ModuleNotFoundError:
-    pass
+from typing import Tuple, List
+import socket
+from contextlib import closing, contextmanager
+import logging
 import validators
+import deepspeed
+from deepspeed.runtime.zero.stage_1_and_2 import estimate_zero2_model_states_mem_needs_all_live;
+from deepspeed.runtime.zero.stage3 import estimate_zero3_model_states_mem_needs_all_live;
+
+
+def find_free_port():
```

Comment:
This function is unused, but remembering a previous conversation it's supposed to find a free port so multiple runs can be executed in the same environment. It might be good to test to see if it's needed or if deepspeed automatically finds a free port too, since I _think_ deepspeed supports that.